__tests__/app.test.js

```diff
@@ -425,3 +424,52 @@ describe("GET /api/users", () => {
   });
 });
 afterAll(() => db.end());
+
+describe('GET /api/articles', () => {
+  it("responds with an array of article objects, sorted by date in descending order", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((res) => {
+        expect(res.body.articles).toBeInstanceOf(Array);
+        expect(res.body.articles).toHaveLength(12);
+        expect(res.body.articles[0]).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          body: expect.any(String),
+          votes: expect.any(Number),
+          topic: expect.any(String),
+          author: expect.any(String),
+          created_at: expect.any(String),
+        });
+        expect(res.body.articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+
+  it("responds with an array of articles filtered by the topic query, sorted by date in descending order", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then((res) => {
+        expect(res.body.articles).toHaveLength(11);
+        expect(res.body.articles[0].topic).toBe("mitch");
+        expect(res.body.articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
```

Comment:
We need to add some more tests:
Status 400. invalid sort_by query, e.g. ?sort_by=bananas
Status 400. invalid order query, e.g. ?order=bananas
Status 404, non-existent topic query, e.g. ?topic=bananas.